ledger/src/blockstore_meta.rs

```diff
@@ -355,167 +356,45 @@ impl ShredIndexV1 {
 ///   requested range, avoiding unnecessary traversal.
 /// - **Simplified Serialization**: The contiguous memory layout allows for efficient
 ///   serialization/deserialization without tree reconstruction.
-#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Default)]
 pub struct ShredIndexV2 {
-    #[serde(with = "serde_bytes")]
-    index: Vec<u8>,
+    index: BitVec<MAX_DATA_SHREDS_PER_SLOT>,
```

Comment:
> Do you have any benchmarks of this vs the current solution?

@alexpyattaev - This PR is moving logic that previously lived in `ShredIndexV2` into an independent type so that `SlotMeta` can utilize the same type. But, the actual changeover for `SlotMeta` will come in a subsequent PR.

Alessandro and/or Zach may have some data handy, but I would not consider providing that data a requirement to ship this PR. As #3900 showed, ser(de) of a `BTreeSet` is pretty rough but even if there were no gains to be had, moving logic into an independent type is good refactoring